src/driver-hidpp20.c

```diff
@@ -1561,6 +1561,8 @@ hidpp20drv_probe(struct ratbag_device *device)
 	drv_data = zalloc(sizeof(*drv_data));
 	ratbag_set_drv_data(device, drv_data);
 	hidpp_device_init(&base, device->hidraw[0].fd);
+	base.hid_reports = (struct hidpp_hid_report*) device->hidraw[0].reports;
+	base.num_hid_reports = device->hidraw[0].num_reports;
```

Comment:
It doesn't make much sense to copy the variable in the struct when they are only used in one single function (`hidpp_get_supported_report_types()`). Please use local variables in `hidpp_get_supported_report_types()` instead if you don't like accessing `device->hidraw[0]` everywhere.

Reply:
Done.